packages/payload/src/uploads/getExternalFile.ts

```diff
@@ -18,27 +24,50 @@ export const getExternalFile = async ({ data, req, uploadConfig }: Args): Promis
       fileURL = `${baseUrl}${url}`
     }
 
-    const headers = uploadConfig.externalFileHeaderFilter
-      ? uploadConfig.externalFileHeaderFilter(Object.fromEntries(new Headers(req.headers)))
-      : { cookie: req.headers.get('cookie')! }
+    let res: NodeFetchResponse | Response
+    try {
+      const headers = uploadConfig.externalFileHeaderFilter
+        ? uploadConfig.externalFileHeaderFilter(Object.fromEntries(new Headers(req.headers)))
```

Comment:
Should/could we extend `externalFileHeaderFilter` to include credentials?
@denolfe

Reply:
This function isn't where this should happen -- this is just a conversation starter about this issue.

Reply:
update: tests are passing using the filtered fetch that does not include credentials by default.

Reply:
Swapped for undici